staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go

```diff
@@ -378,3 +385,186 @@ func (o *ExposeServiceOptions) RunExpose(cmd *cobra.Command, args []string) erro
 	}
 	return nil
 }
+
+func (o *ExposeServiceOptions) createService() (*corev1.Service, error) {
+	if len(o.Selector) == 0 {
+		return nil, fmt.Errorf("selector must be specified")
+	}
+	selector, err := parseLabels(o.Selector)
```

Comment:
Probably worth moving that to `Validate`

Reply:
See comment above